src/engine/source/api/src/catalog/catalog.cpp

```diff
@@ -177,7 +177,16 @@ std::optional<base::Error> Catalog::postResource(const Resource& collection, con
                                         std::get<base::Error>(formatResult).message)};
     }
 
-    const auto contentJson = std::get<json::Json>(formatResult);
+    // Wraps the content in both json and yml
+    auto wrappedContent = json::Json {R"({})"};
```

Comment:
Please don't use the parser constructor, use the default initializer and setObject() method.